### PR TITLE
Hooking package-register updater

### PR DESCRIPTION
**Pull request by adrian-kong**
Calls to create PR in package registry upon release

# JIRA Reference

https://swift-nav.atlassian.net/browse/DEVINFRA-852

### CODE REVIEW

**Review by silverjam — APPROVED** [review]

File: .github/workflows/rust.yaml [hunk 1]
```diff
@@ -83,28 +83,44 @@ jobs:
...
+  registry:
+    if: startsWith(github.ref, 'refs/tags/v')
+    needs: [ build ]
+    name: Publish rust release to package-registry
+    runs-on: ubuntu-latest
+    steps:
+      - name: Hook into package registry
+        uses: swift-nav/swift-cli-publisher@v2
+        env:
+          DL_LINUX: ${{ needs.build.outputs.LINUX_RELEASE_ARCHIVE }}
```
Comment (adrian-kong) [inline .github/workflows/rust.yaml:117]: Seems like we can't access the output of previous jobs if it is dynamically modified (as per https://stackoverflow.com/questions/59175332/using-output-from-a-previous-job-in-a-new-one-in-a-github-action as well)

Comment (silverjam) [reply .github/workflows/rust.yaml:117]: Per discussion we probably can share data by using well known artifact names then uploading/downloading in different jobs, please re-request a review once you have that working

Comment (adrian-kong) [reply .github/workflows/rust.yaml]: Just switched them both to enforce binary targets and hardcoded in the paths

**Comment by silverjam** [comment]
Looks fine, let's do a test release to validate

**Comment by adrian-kong** [comment]
https://github.com/swift-nav/libsbp/runs/7662941672?check_suite_focus=true

https://github.com/swift-nav/package-registry/pull/35

looks to release ok and creates PR,
version is incorrect though

**Review by silverjam — APPROVED** [review]

**Pull request merged**